src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1610,3 +1613,24 @@ export const updatePayments = (
     return okAsync(updatedForm.payments_field)
   })
 }
+
+export const getGlobalBetaFlag = (
+  betaFlag: string,
+): ResultAsync<boolean, DatabaseError> => {
+  return ResultAsync.fromPromise(
+    GlobalBetaModel.findFlag(betaFlag),
+    (error) => {
+      logger.error({
+        message: 'Database error when getting global beta flag status',
+        meta: {
+          action: 'findFlag',
+        },
+        error,
+      })
+
+      return new DatabaseError(
+        `Unable to get global beta flag status. If this issue persists, please submit a Support Form at (${SUPPORT_FORM_LINK})`,
+      )
```

Comment:
Hmm, throwing a runtime error is a bit extreme. I'm thinking if we should return `false`, and make sure we have an alert in place when we detect we cannot read the flags.

Reply:
Decided to keep this DatabaseError and leave the decision of what to return to controllers. Controllers now treat DatabaseErrors the same as returning false, done in [f537622](https://github.com/opengovsg/FormSG/pull/6286/commits/f537622f7ed93d0f01cbc7a4bb0139f0980c45a6).